src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Helpers.cs

```diff
@@ -11,6 +11,12 @@ public static partial class JsonSerializer
     {
         private static TValue? ReadCore<TValue>(JsonConverter jsonConverter, ref Utf8JsonReader reader, JsonSerializerOptions options, ref ReadStack state)
         {
+            if (typeof(T).IsValueType && jsonConverter is ObjectDefaultConverter<TValue> objConverter)
+            {
+                // Fast path for the most popular converter, the cast above is even inlined
+                return objConverter.ReadCore(ref reader, options, ref state);
+            }
```

Comment:
just an experiment, I saw `is JsonConverter` somewhere in hot-paths for TE benchmarks.